stdlib/UUIDs/src/UUIDs.jl

```diff
@@ -94,6 +94,8 @@ function uuid4(rng::AbstractRNG=Random.default_rng())
     UUID(u)
 end
 
+Base._uuid4(rng::AbstractRNG=Random.default_rng()) = uuid4(rng)
```

Comment:
I couldn't find a proper way to "import" stdlib to Base.  Is this an OK approach?

Reply:
Why not just `rand(UInt64)` or something involving `objectid`?

Reply:
It looks to me like most of the uses of `key` could just be replaced with `objectid(::ContextVar)`

Reply:
Actually, the main reason I wanted to use UUID was `uuid5` because I can then use it to get _deterministic_ and namespaced key for "global const" context variables (while `uuid4` is used only for local and "global non-const" context variables):

https://github.com/JuliaLang/julia/blob/ea0e4a7f8da164ba02b2bab612ab73fae97dfbd9/base/contextvariables.jl#L181

(where `join(fullpath, '.')` is something like `"PackageName.SubModuleName.varname"`)

Making the key for "global const" context variable is important for Distributed (otherwise different processes can't agree on the key).  So no, we can't use `objectid`.

Reply:
> I couldn't find a proper way to "import" stdlib to Base. Is this an OK approach?

Yeah it's a tricky problem: you'd like a `uuid4` to use a non-terrible RNG. But we can't pull all of `Random` into Base.

Another workaround for this would be to use `root_module` to look up the `UUIDs` module without loading it. There's precedent for this in the way LibGit2 is looked up in one place:

https://github.com/JuliaLang/julia/blob/97e3fe88d7d3e4f748195a4250b34ed593419b56/base/methodshow.jl#L316

In either case, it's not great to have `Base` missing some functionality when compiled without a stdlib present.

Reply:
I see. Thanks for the explanation. How about vendoring libc-based uuid4 in Base?

```julia
function _uuid4()
    u = reinterpret(UInt128, [Libc.rand() for _ in 1:(sizeof(UInt128) ÷ sizeof(Cint))])[1]
    u &= 0xffffffffffff0fff3fffffffffffffff
    u |= 0x00000000000040008000000000000000
    UUID(u)
end
```

Reply:
What quality of random numbers do we need for this? I guess a large period and good seeding are desired, as the UUIDs may end up in a distributed system? Actually this would suggest that the global RNG from `Random` isn't a great choice for `UUIDs` anyway because the user may control the seeding for reasons unrelated to UUIDs.

Reply:
> Actually this would suggest that the global RNG from `Random` isn't a great choice for `UUIDs` anyway because the user may control the seeding for reasons unrelated to UUIDs.

Oooh, that's a good point. Maybe we should use `Random.RandomDevice()` instead of `Random.default_rng()` in UUIDs? See also https://github.com/JuliaLang/julia/issues/32954#issuecomment-526580532

But I think `_uuid4` above may be OK as the intial implementation since:

* It is used only for `@contextvar local var` and `@contextvar global var` (the latter is available mainly for exploration in REPL and testing).  But the "main entrypoint" is `@contextvar var` which uses uuid5.
* The UUIDs are generated during macro expansion time. So, messing with it requires you to call `Libc.srand(my_seed)` at the top-level of your module. It's totally possible but somewhat unlikely.

In the long run, I think we'd need something like `Base.CoreRandom.RandomDevice`.